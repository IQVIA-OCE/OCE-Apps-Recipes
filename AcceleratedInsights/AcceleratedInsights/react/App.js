import React, { useState, useEffect } from "react";
import { StyleSheet, Linking, Platform, View, Text, useColorScheme } from "react-native";
import { WebView } from "react-native-webview";
import { Button, DarkTheme, DefaultTheme, Loader, Provider as ApolloProvider} from "@oce-apps/apollo-react-native";
import Icon from "react-native-vector-icons/MaterialCommunityIcons";
import { layoutBridge } from "@oce-apps/oce-apps-bridges";
import { queryWithSOQL } from './src/helpers';

Icon.loadFont();

const App = ({ recordId, isTest = false }) => {
  const [uri, setUri] = useState("");
  const [dataFetched, setDataFetched] = useState(false);

  const webview = null;

  let preferredTheme = DefaultTheme;
  let url1 = "";

  const colorScheme = useColorScheme();

  if (Platform.OS !== 'web') {
      preferredTheme = colorScheme === "dark" ? DarkTheme : DefaultTheme;
  }
  else
  {
    preferredTheme = DefaultTheme;
  }

  const handleWebViewNavigationStateChange = (newNavState) => {
    const { url } = newNavState;

    if (!url) return;

    if (url !== uri) {
      this.webview.stopLoading();
      this.webview.goBack();
      Linking.openURL(url);
    }
  };

  const testFuncsOverride2 = () => {
    if (url1 == "https://google.com")
      return true;
    else
      return false;
  };

  const testFuncsOverride3 = () => {
    if (url2 == "https://google.com")
      return true;
    else
      return false;
  };

  const testFuncsOverride = () => {
    testFuncsOverride2();
  };

  const fetchAccountData = async () => {
    if (recordId)
    {
        let query = `select Id, QIDC__OrchestrationLoginTokenLong__c from Account where Id = '${recordId}'`;
        let records = await queryWithSOQL(query);
        if (records.length > 0) {
        setUri(records[0].QIDC__OrchestrationLoginTokenLong__c);
        setDataFetched(true);
        }
    }
  };  

  useEffect(() => {
    fetchAccountData();
  }, []);

  const testFuncs = () => {
    this.webview = WebView;
    handleWebViewNavigationStateChange({ url: "" });
    url1 = "https://google.com";
    url2 = "https://google.com";
    //handleWebViewNavigationStateChange({ url: "https://google.com" });
    testFuncsOverride();
    testFuncsOverride3();
    url1 = "https://google1.com";
    url2 = "https://google1.com";
    testFuncsOverride();
    testFuncsOverride3();
  };

  if (uri)
  {
    if (uri.length > 0 && dataFetched)
    {
      if (Platform.OS === 'web') {
        layoutBridge.setHeight(80);
        return (
          <View style={styles.container}>
            <Text testID="web-test" style={styles.foundText}
            onPress={() => Linking.openURL(uri)}>
              Click here to access to Accelerated Insights.
            </Text>
          </View>
        );
      }
      layoutBridge.setHeight(600);  
      return (
          <ApolloProvider theme={preferredTheme}>
          <WebView
            ref={(ref) => (this.webview = ref)}
            allowsBackForwardNavigationGestures={true}
            source={{ uri: uri }}
            onNavigationStateChange={this.handleWebViewNavigationStateChange}
            testID="webview-test"
            renderLoading={()=>{return(<Loader />)}}
            startInLoadingState={true}
          />
          { isTest &&
            <View>
              <Button
                testID='buttonFuncs-test'
                title="buttonFuncs-test"
                onPress={() => testFuncs()}
              />
            </View>
          }
        </ApolloProvider>
      );
    }
  }
  layoutBridge.setHeight(80);
  return (
    <ApolloProvider theme={preferredTheme}>
      <View style={styles.container}>
        <Text style={styles.notFoundText}>Accelerated Insights not found for this account.</Text>
      </View>
      { isTest &&
        <View>
          <Button
            testID='buttonFuncs-test'
            title="buttonFuncs-test"
            onPress={() => testFuncs()}
          />
        </View>
      }
    </ApolloProvider>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: "center",
    alignItems: "center"
  },
  notFoundText: {
    paddingHorizontal: 15,
    paddingVertical: 25,
    fontWeight: 'bold',
    fontSize: 20,
    color:'red',
  },
  foundText:
  {
    paddingHorizontal: 15,
    paddingVertical: 25,
    fontWeight: 'bold',
    fontSize: 20,
    color:'blue',
  }

});

export default App;

