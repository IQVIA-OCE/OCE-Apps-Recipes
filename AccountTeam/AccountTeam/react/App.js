import React from 'react';
import { NativeModules, ScrollView, Platform } from 'react-native';
import { sfNetAPI } from 'oce-apps-bridges';
import { Provider as ReduxProvider } from 'react-redux';
import { Provider as ApolloProvider } from 'apollo-react-native';
import { AccountTeam } from './src/screens/AccountTeam';
import { store } from './src/store/store';

export default App = ({ instanceId, recordId }) => {

    sfNetAPI.enablePromises();
    if (instanceId) {
        const layout = NativeModules.LayoutBridge;
        layout.setHeight(600, instanceId);
    }
    if (Platform.OS === 'web') {
        return (
            <ReduxProvider store={store}>
                <ApolloProvider>
                    <ScrollView>
                        <AccountTeam accountId={recordId} />
                    </ScrollView>
                </ApolloProvider>
            </ReduxProvider >
        );
    }
    return (
        <ReduxProvider store={store}>
            <ApolloProvider>
                <AccountTeam accountId={recordId} />
            </ApolloProvider>
        </ReduxProvider >
    );
};
