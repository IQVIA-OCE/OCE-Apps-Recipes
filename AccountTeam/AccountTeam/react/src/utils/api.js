import { sfNetAPI } from 'oce-apps-bridges';

export default {
  query: async (soql = '') =>
    new Promise((resolve, reject) => {
      sfNetAPI.query(
        soql,
        data => {
          if (data && data.records) {
            const { records, ...metadata } = data;
            resolve([records, metadata]);
          } else {
            resolve(data);
          }
        },
        e => {
          reject(e);
        }
      );
    }),
  describe: async (objectType = '') =>
    new Promise((resolve, reject) => {
      sfNetAPI.describe(
        objectType,
        data => {
          resolve([data]);
        },
        e => {
          reject(e);
        }
      );
    })
};
