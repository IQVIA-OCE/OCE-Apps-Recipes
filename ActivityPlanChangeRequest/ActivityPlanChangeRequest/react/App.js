import React, { useEffect } from 'react';
import { Provider, useDispatch } from 'react-redux';
import { Portal, Provider as ApolloProvider } from 'apollo-react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import store from './src/store/store';
import UpdateActivityPlan from './src/screens/UpdateActivityPlan/UpdateActivityPlan';
import {
  getAccountInfo,
  getDescribedAPCR,
  getAccountsList,
  getPlanCycleDates,
} from './src/store/applicationSlice/applicationSlice';
import NewActivityPlanChangeRequest from './src/screens/NewActivityPlanChangeRequest/NewActivityPlanChangeRequest';
import RemoveActivityPlan from './src/screens/RemoveActivityPlan/RemoveActivityPlan';
import { createStackNavigator } from 'react-navigation-stack';
import { NEW_ACTIVITY } from './src/constants/routes';
import { createAppContainer } from 'react-navigation';
import NotificationBanner from './src/components/NotificationBanner/NotificationBanner';
import Loader from './src/components/Loader/Loader';
import { NativeModules } from 'react-native';
import AddActivityToPlan from './src/screens/AddActivityToPlan/AddActivityToPlan';

Icon.loadFont();

const MainNavigator = createStackNavigator(
  {
    NewActivityPlanChangeRequest: {
      screen: NewActivityPlanChangeRequest,
      navigationOptions: {
        header: null,
      },
    },
    UpdateActivityPlan: {
      screen: UpdateActivityPlan,
      navigationOptions: {
        header: null,
      },
    },
    RemoveActivityPlan: {
      screen: RemoveActivityPlan,
      navigationOptions: {
        header: null,
      },
    },
    AddActivityToPlan: {
      screen: AddActivityToPlan,
      navigationOptions: {
        header: null,
      },
    },
  },
  {
    initialRouteName: NEW_ACTIVITY,
  }
);
const Navigation = createAppContainer(MainNavigator);

const Root = ({ parentId }) => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getDescribedAPCR());
    dispatch(getAccountInfo(parentId));
    dispatch(getAccountsList());
    dispatch(getPlanCycleDates(parentId));
  }, []);
  return (
    <ApolloProvider>
      <Portal>
        <Loader />
        <NotificationBanner />
      </Portal>
      <Navigation />
    </ApolloProvider>
  );
};

const App = ({ recordId, instanceId }) => {
  if (instanceId) {
    const layout = NativeModules.LayoutBridge;
    layout.setHeight(500, instanceId);
  }

  return (
    <Provider store={store}>
      <Root parentId={recordId} testID="root"/>
    </Provider>
  )
}

export default App;
