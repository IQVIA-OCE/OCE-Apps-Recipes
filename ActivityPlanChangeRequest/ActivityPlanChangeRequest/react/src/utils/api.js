import { sfNetAPI } from 'oce-apps-bridges';

export default {
  query: (soql = '') =>
    sfNetAPI.query(soql).then((data) => {
      if (data && data.records) {
        const { records, ...metadata } = data;
        return [records, metadata];
      } else {
        return data;
      }
    }),
  describe: (objectType = '') =>
    sfNetAPI.describe(objectType).then((data) => [data]),
  create: (objectType = '', fields = {}) =>
    sfNetAPI.create(objectType, fields).then((data) => [data]),
  update: (objectType, id, fields) =>
    sfNetAPI.update(objectType, id, fields).then((data) => [data]),
  del: (objectType, id) => sfNetAPI.del(objectType, id).then((data) => [data]),
  apexRest: (method, endPoint, namespace, params) =>
    sfNetAPI
      .apexRest(method, endPoint, namespace, params)
      .then((data) => [data]),
};
