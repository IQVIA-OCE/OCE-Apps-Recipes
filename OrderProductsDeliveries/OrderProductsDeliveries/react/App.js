/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React, { useEffect, useState } from 'react';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import Header from './src/components/Header/Header';
import OrdersTable from './src/components/OrdersTable/OrdersTable';
import {
  fetchOrderDetailsAsync,
  setRecordInfoAsync,
} from './src/stores/orders';
import { useDispatch, useSelector } from 'react-redux';
import useDebounce from './src/utils/useDebounce';
import { Banner } from 'apollo-react-native';

Icon.loadFont();

const App = ({ recordId }) => {
  const dispatch = useDispatch();
  const { _recordId, _recordType, searchProductValue, brandFilter } = useSelector(
    (state) => state.ordersListReducers
  );
  const notification = useSelector(state => state.notification);

  const debouncedSearchProductValue = useDebounce(searchProductValue, 500);

  useEffect(() => {
    if (recordId) {
      dispatch(setRecordInfoAsync(recordId));
    }
  }, [recordId]);

  useEffect(() => {
    if (_recordId) dispatch(fetchOrderDetailsAsync(_recordType, _recordId));
  }, [_recordId, debouncedSearchProductValue, brandFilter]);

  return (
    <>
      <Banner
        closeIcon
        visible={notification.visible}
        variant={notification.variant}
        icon={notification.icon}
        onCloseIconPress={() => dispatch(closeNotification())}
      >
        {notification.text}
      </Banner>
      <Header />
      <OrdersTable />
    </>
  );
};

export default App;
