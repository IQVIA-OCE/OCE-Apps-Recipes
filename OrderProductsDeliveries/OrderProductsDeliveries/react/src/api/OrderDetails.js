import { DateTime } from 'luxon';
import { environment, databaseManager } from 'oce-apps-bridges';

const NAMESPACE = environment.namespace();

export const fetchOrderDetails = (
  id,
  days,
  searchProductValue,
  brandFilterValueId
) => {
  let idFilter;

  if (id) {
    idFilter = `AND ${NAMESPACE}Account__c = '${id}'`;
  }

  const fromDateTime = DateTime.local()
    .setZone(environment.timeZone())
    .minus({ days })
    .toString();

  let productNameFilter =
    searchProductValue && searchProductValue.length
      ? `AND (${NAMESPACE}Product__r.Name LIKE '%${searchProductValue}%' OR ${NAMESPACE}Product__r.${NAMESPACE}ProductCode__c LIKE '%${searchProductValue}%')`
      : '';

  const fromDateTimeUtc = DateTime.fromISO(fromDateTime, {
    zone: 'utc',
  }).toFormat(`yyyy-MM-dd`);

  const brandFilter =
    brandFilterValueId && brandFilterValueId.length
      ? `AND ${NAMESPACE}Product__r.${NAMESPACE}ParentBrandProductId__c = '${brandFilterValueId}'`
      : '';

  return databaseManager.fetch(
    `SELECT Id, Name, ${NAMESPACE}Account__c, ${NAMESPACE}Free__c, ${NAMESPACE}OrderDelivery2__c, ${NAMESPACE}OrderDelivery2__r.${NAMESPACE}DeliveryDate__c,\
                ${NAMESPACE}OrderLineItem2__c, ${NAMESPACE}ProductName__c, ${NAMESPACE}Product__c, ${NAMESPACE}Product__r.${NAMESPACE}ProductCode__c, ${NAMESPACE}Product__r.Name, ${NAMESPACE}Quantity__c,\
                ${NAMESPACE}Product__r.${NAMESPACE}ParentBrandProductId__r.Name, ${NAMESPACE}Product__r.${NAMESPACE}ParentBrandProductId__c\
        FROM ${NAMESPACE}OrderDetail2__c\
        WHERE ${NAMESPACE}OrderDelivery2__r.${NAMESPACE}DeliveryDate__c >= ${fromDateTimeUtc} AND ${NAMESPACE}OrderDelivery2__r.${NAMESPACE}Status__c NOT IN ('Cancelled') AND ${NAMESPACE}OrderDelivery2__r.${NAMESPACE}Order2__r.${NAMESPACE}Status__c = 'Submitted' ${idFilter} ${productNameFilter} ${brandFilter}\
        ORDER BY ${NAMESPACE}Product__r.Name ASC NULLS FIRST`
  );
};

export const fetchAccounts = (value) => {
  const nameFilter = value ? `AND Name LIKE '%${value}%'` : '';

  return databaseManager.fetch(
    `SELECT Id, Name FROM Account WHERE Name != null ${nameFilter} ORDER BY Name ASC NULLS FIRST Limit 50`
  );
};

export const fetchAccountById = (id) => {
  return databaseManager.fetch(
    `SELECT Id, Name FROM Account WHERE Id = '${id}'`
  );
};

export const fetchOrderById = (id) => {
  return databaseManager.fetch(
    `SELECT Id, Name, ${NAMESPACE}Account__c FROM ${NAMESPACE}Order2__c WHERE Id = '${id}'`
  );
};

export const fetchAllOrderDetailsRecords = async (
  id,
  days,
  searchProductValue,
  brandFilter,
  queryLocator = null
) => {
  try {
    let response;

    if (queryLocator) {
      // Fetch data by queryLocator
      response = await databaseManager.fetch(queryLocator);
    } else {
      // Fetch data by soql query
      response = await fetchOrderDetails(
        id,
        days,
        searchProductValue,
        brandFilter
      );
    }

    // Extract records list from response
    let { records } = response;

    // Is fetch is done or there are other records to fetch
    const { done } = response;

    const nextQueryLocator = response.queryLocator;

    // If fetch is not done, fetch data by queryLocator and merge its output into the records array recursively

    // ToDo: Should check done field instead queryLocator
    // Related to totalSize: 0 issue
    if (nextQueryLocator) {
      const extraData = await fetchAllOrderDetailsRecords(
        id,
        days,
        searchProductValue,
        brandFilter,
        nextQueryLocator
      );
      records = records.concat(extraData);
    }
    // Else return the complete array of records
    return records;
  } catch (err) {
    return console.log(err);
  }
};
