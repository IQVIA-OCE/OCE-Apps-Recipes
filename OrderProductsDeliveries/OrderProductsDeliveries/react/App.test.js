import { render, waitFor } from '@testing-library/react-native';
import React from 'react';
import App from './App';
import { useSelector, useDispatch } from 'react-redux';
import { fetchAccounts } from './src/api/OrderDetails';

jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock('./src/api/OrderDetails', () => ({
    fetchAccounts: jest.fn(),
}));

const accounts = [
    {
        Id: '123',
        Name: '321'
    }
]

it('App', async () => {
    useDispatch.mockReturnValue(() => jest.fn());
    fetchAccounts.mockResolvedValue({
        records: accounts
    })
    useSelector.mockImplementation((cb) => cb({
        ordersListReducers: {
            searchProductValue: '',
            brandOptions: [],
            _recordId: '123'
        },
        notification: {}
    }));
    const { container, toJSON } = render(<App />);
    await waitFor(() => fetchAccounts);
    expect(toJSON(container)).toMatchSnapshot();
})