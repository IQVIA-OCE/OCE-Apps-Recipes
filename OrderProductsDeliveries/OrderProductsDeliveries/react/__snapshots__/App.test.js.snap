// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `
Array [
  <View
    collapsable={false}
    nativeID="animatedComponent"
    style={
      Object {
        "backgroundColor": "#ffffff",
        "elevation": 1,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 0.75,
        "zIndex": 99,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "overflow": "hidden",
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "height": 0,
          }
        }
      />
      <View
        collapsable={false}
        nativeID="animatedComponent"
        onLayout={[Function]}
        style={
          Object {
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "translateY": -0,
              },
            ],
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "position": "absolute",
              "right": 5,
              "top": 4,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            centered={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            nativeID="animatedComponent"
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            style={Object {}}
          >
            <View
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 15,
                      "height": 30,
                      "width": 30,
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "#595959",
                        "fontSize": 20,
                      },
                      Array [
                        Object {
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 0,
              "marginLeft": 16,
              "marginRight": 56,
              "marginTop": 8,
            }
          }
        >
          <View
            style={
              Object {
                "maxWidth": "100%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "margin": 8,
                    },
                    Object {
                      "color": "#000000",
                    },
                  ],
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "margin": 4,
            }
          }
        />
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 32,
                "marginVertical": 2,
              },
              undefined,
            ],
            Array [
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 32,
                "marginVertical": 2,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      Deliveries
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingTop": 15,
          "width": "100%",
        }
      }
    >
      <View
        collapsable={false}
      >
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "width": 176,
              },
              Array [
                Object {
                  "width": "100%",
                },
                undefined,
                Object {
                  "width": 300,
                },
              ],
              Object {
                "opacity": 1,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "#D9D9D9",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
                Object {
                  "width": 300,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 40,
                  "minHeight": 40,
                  "overflow": "hidden",
                  "paddingBottom": 0,
                  "paddingTop": 0,
                }
              }
            >
              <View
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "left": 10,
                    "opacity": 1,
                    "position": "absolute",
                    "right": "auto",
                    "top": "50%",
                    "transform": Array [
                      Object {
                        "translateY": -12.5,
                      },
                    ],
                    "zIndex": 5,
                  }
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "#999",
                        "fontSize": 25,
                      },
                      Array [
                        Object {
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <TextInput
                editable={true}
                maxResizableWidth={750}
                minHeight={60}
                minWidth={0}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onSubmitEditing={[Function]}
                paddingLeft={40}
                paddingRight={8}
                placeholder="Account Name"
                placeholderTextColor="rgba(0, 0, 0, 0.54)"
                selectionColor="#0768FD"
                style={
                  Array [
                    Object {
                      "margin": 0,
                      "textAlign": "left",
                      "zIndex": 5,
                    },
                    undefined,
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "height": 40,
                      "minHeight": 0,
                      "paddingLeft": 40,
                      "paddingRight": 8,
                      "paddingVertical": 5,
                      "textAlignVertical": "center",
                      "width": "100%",
                    },
                  ]
                }
                testID="search"
                value=""
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "width": 300,
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "width": 176,
              },
              Array [
                Object {
                  "width": "100%",
                },
                undefined,
                undefined,
              ],
              Object {
                "opacity": 1,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "#D9D9D9",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
                Object {
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 40,
                  "minHeight": 40,
                  "overflow": "hidden",
                  "paddingBottom": 0,
                  "paddingTop": 0,
                }
              }
            >
              <View
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "left": 10,
                    "opacity": 1,
                    "position": "absolute",
                    "right": "auto",
                    "top": "50%",
                    "transform": Array [
                      Object {
                        "translateY": -12.5,
                      },
                    ],
                    "zIndex": 5,
                  }
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "#999",
                        "fontSize": 25,
                      },
                      Array [
                        Object {
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <TextInput
                editable={true}
                maxResizableWidth={750}
                minHeight={60}
                minWidth={0}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                paddingLeft={40}
                paddingRight={8}
                placeholder="Search product name"
                placeholderTextColor="rgba(0, 0, 0, 0.54)"
                selectionColor="#0768FD"
                style={
                  Array [
                    Object {
                      "margin": 0,
                      "textAlign": "left",
                      "zIndex": 5,
                    },
                    undefined,
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "height": 40,
                      "minHeight": 0,
                      "paddingLeft": 40,
                      "paddingRight": 8,
                      "paddingVertical": 5,
                      "textAlignVertical": "center",
                      "width": "100%",
                    },
                  ]
                }
                testID="search-view"
                value=""
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "width": 300,
          }
        }
        testID="filterWrapper"
      >
        <View
          style={
            Array [
              Object {
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          <View>
            <View
              collapsable={false}
            >
              <View
                onLayout={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    },
                    Object {
                      "backgroundColor": "rgba(246, 246, 246, 0.4)",
                      "borderColor": "rgba(217, 217, 217, 0.4)",
                      "borderRadius": 4,
                      "borderWidth": 1,
                      "elevation": 0,
                      "height": 40,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "stretch",
                      "flex": 1,
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "paddingLeft": 8,
                    }
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    hitSlop={
                      Object {
                        "bottom": 14,
                        "left": 14,
                        "right": 14,
                        "top": 14,
                      }
                    }
                    nativeID="animatedComponent"
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "flex": 1,
                        "opacity": 1,
                      }
                    }
                  >
                    <Text
                      numberOfLines={1}
                      style={
                        Array [
                          Object {
                            "color": "rgba(0, 0, 0, 0.4)",
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": "400",
                          },
                        ]
                      }
                    >
                      Brand
                    </Text>
                  </View>
                </View>
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "disabled": true,
                    }
                  }
                  accessible={true}
                  centered={true}
                  collapsable={false}
                  focusable={true}
                  hitSlop={
                    Object {
                      "bottom": 6,
                      "left": 6,
                      "right": 6,
                      "top": 6,
                    }
                  }
                  nativeID="animatedComponent"
                  onClick={[Function]}
                  onLayout={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  pointerEvents="box-only"
                  style={Object {}}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "alignItems": "center",
                            "justifyContent": "center",
                            "margin": 6,
                            "overflow": "hidden",
                          },
                          Object {
                            "borderRadius": 18,
                            "height": 36,
                            "width": 36,
                          },
                          Object {
                            "opacity": 0.32,
                          },
                          Object {
                            "margin": 0,
                          },
                        ],
                      ]
                    }
                  >
                    <View>
                      <Text
                        accessibilityElementsHidden={true}
                        allowFontScaling={false}
                        importantForAccessibility="no-hide-descendants"
                        pointerEvents="none"
                        style={
                          Array [
                            Object {
                              "color": "rgba(0, 0, 0, 0.4)",
                              "fontSize": 24,
                            },
                            Array [
                              Object {
                                "transform": Array [
                                  Object {
                                    "scaleX": 1,
                                  },
                                ],
                              },
                              Object {
                                "backgroundColor": "transparent",
                              },
                            ],
                            Object {
                              "fontFamily": "Material Design Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                      >
                        
                      </Text>
                    </View>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "bottom": 0,
                          "left": 0,
                          "overflow": "hidden",
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        },
                        Object {
                          "borderRadius": 0,
                        },
                      ]
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 400,
        "justifyContent": "center",
      }
    }
  >
    <Text>
      No data found
    </Text>
  </View>,
]
`;
